src/validators.rs

```diff
@@ -1,4 +1,5 @@
 use crate::*;
+use chrono::{DateTime, Utc};
```

Comment:
how much bloat does chrono add? This library is used in some space constrained environments? Can we do without?

Reply:
Wouldn't it be compiled out in those environments? I've done a few reward endpoints on my api branch and its really the thing to use for the timestamp parsing.

If we're not convinced it'll be compiled out, I would suggest feature gating...

Reply:
This adds 59k to release. If you want to reduce file size you should consider adding: 
```toml
[profile.release]
opt-level = 'z'
lto = true
codegen-units = 1
```
To `Cargo.toml` that would shave off 27k.

Since there are data range queries and if I understand correctly, we would continue to add more endpoints into this lib, it would be very useful to have proper datetime type. Do we intend this to be a full api client?

Reply:
> Wouldn't it be compiled out in those environments? I've done a few reward endpoints on my api branch and its really the thing to use for the timestamp parsing.
> 
> If we're not convinced it'll be compiled out, I would suggest feature gating...

it won't be compiled out of the client uses it

Reply:
> Since there are date range queries and if I understand correctly, we would continue to add more endpoints into this lib, it would be very useful to have proper datetime type. Do we intend this to be a full api client?

Yes, but strings vs actual date/time doesn't make it more or less full, just more or less typed, right?

Reply:
I mean as a consumer of this create, it would be a lot less friction if datetime could be treated as such for any kind of calculation, sorting, or whatever the consumer may want to do. Otherwise consumers would have to always parse the string into a datetime format wherever they needed so in code. That "feels" a lot less ergonomic compared to other apis. It is very commonly used in other api client crates like s3 ones, spotify ones, etc.

Reply:
It's fine to have it btw.. I was just wanting to have the discussion

Reply:
> I mean as a consumer of this create, it would be a lot less friction if datetime could be treated as such for any kind of calculation, sorting, or whatever the consumer may want to do. Otherwise consumers would have to always parse the string into a datetime format wherever they needed so in code. That "feels" a lot less ergonomic compared to other apis. It is very commonly used in other api client crates like s3 ones, spotify ones, etc.

yup agreed, thanks for the 'splaining